subsys/debug/tracing/tracing_backend_posix.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <stdio.h>
+#include <kernel.h>
+#include <assert.h>
+#include <cmdline.h>
+#include <tracing_backend.h>
+
+static void *out_stream;
+static const char *file_name;
```

Comment:
How is supposed to set `file_name` ? The scope of this variable is this file and there is no API to change it.

Reply:
```
void tracing_backend_posix_option(void)
{
        static struct args_struct_t tracing_backend_option[] = {
                {
                        .manual = false,
                        .is_mandatory = false,
                        .is_switch = false,
                        .option = "trace-file",
                        .name = "file_name",
                        .type = 's',
                        .dest = (void *)&file_name,
                        .call_when_found = NULL,
                        .descript = "File name for tracing output.",
                },
                ARG_TABLE_ENDMARKER
        };

        native_add_command_line_opts(tracing_backend_option);
}

```
native_add_command_line_opts will parse tracing_backend_option which contains the address of file_name.

Reply:
my point is, how the user change the value of `file_name` ?  The struct `tracing_backend_option` is statically defined with the address of `file_name` that is constant.

Reply:
@ceolin file_name will point to const string passed by user(when run ./zephyr.elf channel0_1), it can point another string(file name), but we don't have that case, all the tracing data will only be save one file.

Reply:
got it !